resources/testharness.js

```diff
@@ -1184,19 +1184,21 @@ policies and contribution forms [3].
                {actual:actual});
         assert(actual.length === expected.length,
                "assert_array_equals", description,
-               "lengths differ, expected ${expected} got ${actual}",
-               {expected:expected.length, actual:actual.length});
+               "lengths differ, expected array ${expected} got ${actual}",
```

Comment:
The original reason for not doing this is concern about what happens when the array is large e.g. if you have something like `actual = []; actual[10000]=1`. So if we are going to print the array values, I'd like a helper function that checks if the array length is under some limit and if so prints the while thing, but otherwise just a limited number of elements close to an index (e.g. we might output full arrays up to 20 elements and then at most 20 elements in this case the last 20 elements, or something).